### ISSUE TITLE
the performance of prm800k

### ISSUE THREAD

**Issue by aojunzz** [scottlogic-alex/prm800k-denorm#1]
Hi, 

Thanks for your great works, do you try to train the pretrained model (e.g., llama v1 or v2) using the prm800k datasets.

**Comment by scottlogic-alex** [comment]
yes, we used the [solutions-only](https://huggingface.co/datasets/sl-alex/openai-prm800k-solutions-only) distribution, to train a QLoRA adapter over LLaMAv1-13b. we also resized the input+output embeddings to teach it new special tokens:  
https://huggingface.co/sl-alex/llama-13b-alpaca-stepwise-lora-embtuned

it did indeed learn the new conversational style (stepwise reasoning) that we taught it (it indicates its steps explicitly), and it did generalise to answering non-maths problems via stepwise reasoning. the answers seem longer than Alpaca's, which is sometimes terse. but it sometimes uses the step indicators incorrectly, e.g. to indicate turns in a two-person conversation.

we haven't done a rigorous evaluation of it, but when we ask it questions manually: it's still not super clever. the reasoning process is coherent, and often funny, but wrong. but this could easily be because 13b wasn't a genius to begin with.

it does seem to reward-hack quite a lot (it prefers to make long, many-step responses, which don't often terminate in an answer).

to be honest, I think the maths problems in the dataset are really challenging, and not enough examples of the same types of problem. I think it's a good dataset for learning the stepwise reasoning style, but I don't think it'll learn to get better at those maths problems.

if you're interested in (elementary) _maths_ performance: it's worth taking a look at OpenAI's prior work, [GSM8K](https://huggingface.co/datasets/gsm8k), too, for a dataset with simpler maths problems (and toolformer-style annotations). it's stepwise too, although not indicated as explicitly.
